tools/build-id.py

```diff
@@ -0,0 +1,122 @@
+import zipfile
+import argparse
+
+READELF_TOOL_PATH="/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-readelf"
+
+FLAVORS = ['base', 'objectServer']
+ARCHS = ['arm64-v8a', 'armeabi-v7a', 'x86', 'x86_64']
+
+def file_exists(path):
+    f = None
+    try:
+        f = open(path)
+        return True
+    except IOError:
+        return False
+    finally:
+        if f:
+            f.close()
+
+def find_ndk():
+    PROPS_FILE=f"{os.getcwd()}/../realm/local.properties"
+    if not file_exists(PROPS_FILE):
+        print(f"{PROPS_FILE} not found! NDK location cannot be determined")
+        exit()
+    
+    return subprocess.check_output(f'grep "ndk.dir" "{PROPS_FILE}" | cut -d = -f2', shell=True).decode('ascii').strip()
+
+
+BUILD_ID_EXTRACT_COMMMAND = find_ndk() + READELF_TOOL_PATH + ' -n {0} | grep "Build ID" | cut -d ":" -f2'
+
+def download_file(url, path):
+    urllib.request.urlretrieve(url, path)
+
+
+def generator(major_range, minor_range, patch_range):
+    for major in major_range:
+        for minor in minor_range:
+            for patch in patch_range:
+                yield f"{major}.{minor}.{patch}"
+
+
+
+
+
+def get_path(path):
+    try:
+        os.mkdir(path)
+    except OSError:
+        pass
+    return path
+
+
+def unzip(zip_file, path):
+    with zipfile.ZipFile(zip_file,"r") as zip_ref:
+        zip_ref.extractall(path)
+
+def format_table_line(build_id, version, flavor, arch):
+    return f"{build_id.ljust(41)} : {version.ljust(10)} {flavor.ljust(14)} {arch}"
+
+
+def extract_build_ids(path, version):
+    for flavor in FLAVORS:
+        for arch in ARCHS:
+            so_path = f"{path}{version}/{flavor}/{arch}/librealm-jni.so"
+            build_id = subprocess.check_output(BUILD_ID_EXTRACT_COMMMAND.format(so_path), shell=True).strip().decode('ascii')
+            yield format_table_line(build_id, version, flavor, arch)
+
+def parse_range(version):
+    if version is None:
+        print("⚠️ Please define major minor and patch, see --help ")
+        exit()
+
+    if '-' in version:
+        version_range = version.split('-')
+        return range(int(version_range[0]), int(version_range[1]) + 1)
+    
+    return [int(version)]
+
+if __name__ == "__main__":
+    print(BUILD_ID_EXTRACT_COMMMAND)
+    parser = argparse.ArgumentParser(description='Extract build ids out from S3.')
+    parser.add_argument('major', nargs='?', help='specific version: 1 or range: 1-3')
+    parser.add_argument('minor', nargs='?', help='specific version: 1 or range: 1-3')
+    parser.add_argument('patch', nargs='?', help='specific version: 1 or range: 1-3')
+
+    args = parser.parse_args()
+
+    major_range = parse_range(args.major)
+    minor_range = parse_range(args.minor)
+    patch_range = parse_range(args.patch)
+    
+    zips_path = get_path(os.getcwd() + "/downloads/")
+    libs_path = get_path(os.getcwd() + "/libs/")
+    build_ids = []
+
+    print("-- This process can take some minutes ⏰ --")
```

Comment:
Ideally out should keep status information and program output separate. So maybe write all the debug info to stderr. This makes it way easier to use the tool to generate processable files.